src/components/screens/UserProfile/UserPostsScreen.tsx

```diff
@@ -10,21 +11,51 @@ import LoadingErrorView from "../../common/Loading/LoadingErrorView";
 import NotFoundView from "../../common/Loading/NotFoundView";
 import RefreshControl from "../../common/RefreshControl";
 
+interface IRouteParams {
+  fullUsername?: string;
+  isSavedPosts?: boolean;
+}
+
 interface IProps {
-  route: any;
+  route: Route<"UserPostsScreen", IRouteParams>;
 }
 
 function UserPostsScreen({ route }: IProps) {
   const profile = useProfile(false, route?.params?.fullUsername);
 
   const theme = useTheme();
 
+  const isSavedPosts = React.useMemo(
+    () => route?.params?.isSavedPosts || false,
+    [route]
+  );
+
+  const posts = React.useMemo(
```

Comment:
I think we want to re-render whenever posts changes because whenever upvoting/downvoting or saving posts, we update the posts state.

Reply:
This will cause an update to the posts object anyway which is a dependency of the hook thus forcing it to update its own return state.